### PR TITLE
Add templates for READMEs.

### PR DESCRIPTION
**Pull request by tswast**
This uses https://github.com/jreese/markdown-pp to include markdown from
other files.  The templates in this repository can also be used in other
repositories with the `!INCLUDEURL` directive.

@lesv PTAL

### CODE REVIEW

**Comment by tswast** [comment]
I cancelled the Travis build since this is just changes to a README.

File: .github/templates/appengine-flexible-deploy-gcloud.mdpp [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+To run the application locally, use the Maven Jetty plugin.
+
+    mvn clean jetty:run
+
+View the app at [localhost:8080](http://localhost:8080).
+
+To deploy the app to App Engine, run
+
+    mvn clean gcloud:deploy
```
Comment (lesv) [inline .github/templates/appengine-flexible-deploy-gcloud.mdpp]: mvn appengine:deploy

Comment (lesv) [inline .github/templates/appengine-flexible-deploy-gcloud.mdpp]: Ah - we have choices.

File: flexible/README.md [hunk 2]
```diff
@@ -9,44 +9,50 @@ scaffolding for other frameworks and use cases.
...
+    mvn clean jetty:run
 
-   ```
-   mvn clean gcloud:deploy
-   ```
+View the app at [localhost:8080](http://localhost:8080).
+
+To deploy the app to App Engine, run
+
+    mvn clean gcloud:deploy
```
Comment (lesv) [inline flexible/README.md]: appengine:deploy

File: flexible/README.mdpp [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+repos](https://github.com/GoogleCloudPlatform) for sample applications and
+scaffolding for other frameworks and use cases.
+
+[aeflex-docs]: https://cloud.google.com/appengine/docs/flexible/
+
+!INCLUDE "../.github/templates/getting-sample-code.mdpp"
+
+!INCLUDE "../.github/templates/appengine-flexible-before-you-begin.mdpp"
+
+!INCLUDE "../.github/templates/appengine-flexible-deploy-gcloud.mdpp"
```
Comment (lesv) [inline flexible/README.mdpp]: Does this actually work?

**Comment by lesv** [comment]
Lets only talk about jetty:run-exploded, appengine:run & appengine:deploy

File: flexible/README.mdpp (hunk 3 above)
Comment (tswast) [inline flexible/README.mdpp]: Yep!

It generate the README.md file when I run the markdown-pp command against it.

I wrote some instructions here: https://github.com/GoogleCloudPlatform/java-docs-samples/blob/fdff8960273a8b9fb4efbd3de1a736d8db380fe9/.github/templates/README.md

File: .github/templates/appengine-flexible-deploy-gcloud.mdpp (hunk 1 above)
Comment (tswast) [inline .github/templates/appengine-flexible-deploy-gcloud.mdpp]: Removed this file. We should update our samples to use the new plugins soon, then.

**Pull request merged**